typescript/cli/src/tests/warp/warp-deploy.e2e-test.ts

```diff
@@ -0,0 +1,540 @@
+
+      const [expectedTokenSymbol, expectedTokenDecimals] = await Promise.all([
+        token.symbol(),
+        token.decimals(),
+      ]);
+      const COMBINED_WARP_CORE_CONFIG_PATH = getCombinedWarpRoutePath(
+        expectedTokenSymbol,
+        [CHAIN_NAME_2, CHAIN_NAME_3],
+      );
+
+      const warpConfig: WarpRouteDeployConfig = {
+        [CHAIN_NAME_2]: {
+          type: TokenType.collateral,
+          token: token.address,
+          mailbox: chain2Addresses.mailbox,
+          owner: ownerAddress,
+        },
+        [CHAIN_NAME_3]: {
+          type: TokenType.synthetic,
+          mailbox: chain3Addresses.mailbox,
+          owner: ownerAddress,
+        },
+      };
+
+      writeYamlOrJson(WARP_DEPLOY_OUTPUT_PATH, warpConfig);
+
+      const steps: TestPromptAction[] = [
+        {
+          check: (currentOutput) =>
+            currentOutput.includes('Please enter the private key for chain'),
+          input: `${ANVIL_KEY}${KeyBoardKeys.ENTER}`,
+        },
+        {
+          check: (currentOutput) =>
+            currentOutput.includes('Please enter the private key for chain'),
+          input: `${ANVIL_KEY}${KeyBoardKeys.ENTER}`,
+        },
+        {
+          check: (currentOutput) =>
+            currentOutput.includes('Is this deployment plan correct?'),
+          input: KeyBoardKeys.ENTER,
+        },
+      ];
+
+      // Deploy
+      const output = hyperlaneWarpDeployRaw({
+        warpCorePath: WARP_DEPLOY_OUTPUT_PATH,
+      })
+        .stdio('pipe')
+        .nothrow();
+
+      const finalOutput = await handlePrompts(output, steps);
+
+      // Assertions
+      expect(finalOutput.exitCode).to.equal(0);
+      const updatedWarpDeployConfig_2 = await readWarpConfig(
+        CHAIN_NAME_2,
+        COMBINED_WARP_CORE_CONFIG_PATH,
+        WARP_DEPLOY_OUTPUT_PATH,
+      );
+
+      expect(updatedWarpDeployConfig_2[CHAIN_NAME_2].type).to.equal(
+        TokenType.collateral,
+      );
+      expect(updatedWarpDeployConfig_2[CHAIN_NAME_2].decimals).to.equal(
+        expectedTokenDecimals,
+      );
+      expect(updatedWarpDeployConfig_2[CHAIN_NAME_2].symbol).to.equal(
+        expectedTokenSymbol,
+      );
+      expect(updatedWarpDeployConfig_2[CHAIN_NAME_2].mailbox).to.equal(
+        chain2Addresses.mailbox,
+      );
+
+      const updatedWarpDeployConfig_3 = await readWarpConfig(
+        CHAIN_NAME_3,
+        COMBINED_WARP_CORE_CONFIG_PATH,
+        WARP_DEPLOY_OUTPUT_PATH,
+      );
+
+      expect(updatedWarpDeployConfig_3[CHAIN_NAME_3].type).to.equal(
+        TokenType.synthetic,
+      );
+      expect(updatedWarpDeployConfig_3[CHAIN_NAME_3].decimals).to.equal(
+        expectedTokenDecimals,
+      );
+      expect(updatedWarpDeployConfig_3[CHAIN_NAME_3].symbol).to.equal(
+        expectedTokenSymbol,
+      );
+      expect(updatedWarpDeployConfig_3[CHAIN_NAME_3].mailbox).to.equal(
+        chain3Addresses.mailbox,
+      );
+    });
```

Comment:
when I see blocks like this I expect at least a little effort to dedupe/be concise
obviously these are just tests but I think maintainability matters here too